snuba/migrations/validator.py

```diff
@@ -0,0 +1,199 @@
+import re
+from typing import Sequence, Union
+
+from snuba.clickhouse.native import ClickhousePool
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.migrations.migration import ClickhouseNodeMigration
+from snuba.migrations.operations import AddColumn, CreateTable, DropColumn, SqlOperation
+
+ENGINE_REGEX = r"^Distributed(\(.+\))$"
+
+
+class InvalidMigrationOrderError(Exception):
+    """
+    Raised when a migration ops are not in the correct order.
+    """
+
+    pass
+
+
+class NoDistClusterNodes(Exception):
+    """
+    The migration has no distributed cluster nodes, so we can't validate the order.
+    """
+
+    pass
+
+
+class DistributedEngineParseError(Exception):
+    """
+    Cant parser the distributed engine string value
+    """
+
+    pass
+
+
+def validate_migration_order(migration: ClickhouseNodeMigration) -> None:
```

Comment:
How do you plan to run this validation? During unit tests or when you try to execute the migration in the distributed cluster? The second would be too late. At that point the migration will already be merged and we will not be able to revert it.

Reply:
In unit tests. I have a test here https://github.com/getsentry/snuba/blob/8f986081747206bbac5d12655a0271917c0ef5b0/tests/migrations/test_validator.py#L25-L47 that validates all migrations.